### PR TITLE
feat(bridge): add minting cli

### PR DESCRIPTION
**Pull request by shkangr**
Can transfer WNCG using CLI like example below

first argument is address ( destination )
second argument is amount ( to transfer )
third argument is gasPrice and it's optional

How to test CLI
```
case1:
npm run mint 0x954941eC7FACf9A81e2f026A356fF83F54a5827F 0.001 50
transfer to 0x954941eC7FACf9A81e2f026A356fF83F54a5827F
amount: 0.001
gasPrice: 50
```

```
case2:
npm run mint 0x954941eC7FACf9A81e2f026A356fF83F54a5827F 123
transfer to 0x954941eC7FACf9A81e2f026A356fF83F54a5827F
amount: 123
gasPrice: 50 ( if env variable "gasPrice" doesn't exist )
```

### CODE REVIEW

File: bridge/Dockerfile [hunk 1]
```diff
@@ -11,5 +11,7 @@ FROM node:16.17-alpine
 
 COPY --from=build-env /build/dist /app
 COPY --from=build-env /build/node_modules /app/node_modules
+COPY package*.json /app
```
[the remark this answers is not in the document]
Comment (shkangr) [reply bridge/Dockerfile:14]: I just wanted to copy package.json and other files start with package.. ( ex. package-lock.json ) if exists.
But there is only package.json file at bridge directory, so I'm gonna fix it.

File: bridge/scripts/mint-script.ts [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+    ]);
+
+    const safeTransactionData: SafeTransactionDataPartial = {
+        to: WNCG_CONTRACT_ADDRESS,
+        value: "0",
+        data,
+        gasPrice,
+    };
+
+    // Create a Safe transaction with the provided parametersx
```
Comment (ahramyun7) [inline bridge/scripts/mint-script.ts:128]: `parametersx` seems mis-typed. Can you update the comment?

File: bridge/scripts/mint-script.ts [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+const txServiceUrl = process.env.SAFE_TX_SERVICE_URL!;
+const safeService = new SafeServiceClient({
+    txServiceUrl,
+    ethAdapter: ethAdapterOwner1,
+});
+let safeSdkOwner1: Safe;
+let safeAddress: string;
+
+// If you have an existing Safe, you can use it instead of deploying a new one
+// https://app.safe.global/home?safe=eth:0x4A2FbE06004e37dE6Fe7Da59a53D14a407Def0ed
```
Comment (ahramyun7) [inline bridge/scripts/mint-script.ts:58]: I think it's not a good idea to keep this comment, (Why ETH Mainnet Safe Address here?) could be better to just remove it.

File: bridge/scripts/mint-script.ts [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+        safeTransactionData: safeTransaction.data,
+        safeTxHash,
+        senderAddress: await owner1Signer.getAddress(),
+        senderSignature: senderSignature.data,
+    });
+    await safeService.proposeTransaction({
+        safeAddress: ethers.utils.getAddress(safeAddress),
+        safeTransactionData: safeTransaction.data,
+        safeTxHash,
+        senderAddress: ethers.utils.getAddress(owner1SignerAddress),
```
Comment (ahramyun7) [inline bridge/scripts/mint-script.ts:151]: L144 `senderAddress` uses `senderAddress: await owner1Signer.getAddress()` L151 `senderAddress` uses `senderAddress: ethers.utils.getAddress(owner1SignerAddress)`.

also `safeAddress: ethers.utils.getAddress(safeAddress)` is different compared to the log.

I think it's better to use same format, even though it's insignificant, always better to have same value for actual code and LOG message.

File: bridge/scripts/mint-script.ts [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+            console.log("Cancel Minting....");
+            process.exit(1);
+        }
+
+        console.log(`Trnasfer WNCG to ${destination}, amount: ${amount} ...`);
+
+        await initalizeSafe();
+
+        const recipient = destination;
+        const gasPrice = 40;
```
Comment (ahramyun7) [inline bridge/scripts/mint-script.ts:274]: maybe change the gasPrice to 50, so that it could be more safe, based on : https://milkroad.com/ethereum/gas

Comment (longfin) [reply bridge/scripts/mint-script.ts:274]: I think that it needed to be configurable (by command line args or env)

File: bridge/scripts/mint-script.ts (hunk 4 above)
Comment (shkangr) [reply bridge/scripts/mint-script.ts]: fixed!
set log for same value ( checkSummed address )

File: bridge/scripts/mint-script.ts (hunk 3 above)
Comment (shkangr) [reply bridge/scripts/mint-script.ts]: removed comments which include contract address
fixed!

File: bridge/scripts/mint-script.ts (hunk 2 above)
Comment (shkangr) [reply bridge/scripts/mint-script.ts]: fixed!

File: bridge/scripts/mint-script.ts (hunk 5 above)
Comment (shkangr) [reply bridge/scripts/mint-script.ts]: @ahramyun7 @longfin 

I added optional CLI argument "gasPrice".

If it(=gasPrice) is not came from cli arguments, 
First: value of gasPrice is set to process.env.gasPrice
Second: If env variable (=gasPrice) is not exists, then value of gasPrice is set to default( =50 )

**Review by longfin — APPROVED** [review]

**Comment by ahramyun7** [comment]
`npm run mint 0x954941eC7FACf9A81e2f026A356fF83F54a5827F 0.0001 hello`

`>> Are you trying to Transfer WNCG. To: 0x954941eC7FACf9A81e2f026A356fF83F54a5827F, Amount: 0.0001, gasPrice: NaN ??`

If you put a string in gasPrice, it comes out as gasPrice NaN, but it would be nice to handle this case as well.

**Comment by shkangr** [comment]
@ahramyun7 added validation check for gasPrice, if it's not a number type, exit process.

This is the test case I did after fixed
```
npm run mint 0x954941eC7FACf9A81e2f026A356fF83F54a5827F 0.0001 hello

gasPrice:hello is not a number
Cancel Minting...
```

**Review by ahramyun7 — APPROVED** [review]
gasPrice validation check confirmed.

LGTM. Approved!

File: bridge/scripts/mint-script.ts [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+
+    if (Number.isNaN(gasPriceToNum)) {
+        console.log(`gasPrice:${gasPrice} is not a number`);
+        console.log("Cancel Minting...");
+        process.exit(1);
+    }
+
+    // Validation Check of Transfer Amount
+    if (
+        amountToNum > Number(process.env.MAX_TRANSFER_AMOUNT) ||
```
Comment (moreal) [inline bridge/scripts/mint-script.ts:243]: *Not critical to merge*

----------

The `MAX_TRANSFER_AMOUNT` environment variable seems like a new one. It should be documented.

File: bridge/scripts/mint-script.ts [hunk 7]
```diff
@@ -0,0 +1,300 @@
...
+        console.log("Invalid Amount Input");
+        console.log("Cancel Minting...");
+        process.exit(1);
+    }
+
+    try {
+        console.log("destination??", destination);
+        ethers.utils.getAddress(destination);
+    } catch (e) {
+        console.error(e);
```
Comment (moreal) [inline bridge/scripts/mint-script.ts:255]: It seems better to let users know the error was shown while what context.

```suggestion
        console.error("While checking the destination address, error occurred: ", e);
```

File: bridge/scripts/mint-script.ts [hunk 8]
```diff
@@ -0,0 +1,300 @@
...
+    await sleep(1);
+    console.clear();
+
+    console.log(
+        "==========================================================================================================="
+    );
+    console.log(
+        `Are you trying to Transfer WNCG. To: ${destination}, Amount: ${amount}, gasPrice: ${gasPriceToNum} ??`
+    );
+    console.log('If Correct, Enter "yes", If not Enter anything');
```
Comment (moreal) [inline bridge/scripts/mint-script.ts:269]:
```suggestion
    console.log('If Correct, Enter "yes". If not, Enter anything');
```

File: bridge/scripts/mint-script.ts [hunk 9]
```diff
@@ -0,0 +1,300 @@
...
+    await reader.on("line", async (input) => {
+        reader.close();
+
+        if (input.toLowerCase() !== "yes") {
+            console.log("Cancel Minting....");
+            process.exit(1);
+        }
+
+        console.log(
+            `Trnasfer WNCG to ${destination}, amount: ${amount}, gasPrice: ${gasPriceToNum} ...`
```
Comment (moreal) [inline bridge/scripts/mint-script.ts:285]: Just typo.

```suggestion
            `Transfer WNCG to ${destination}, amount: ${amount}, gasPrice: ${gasPriceToNum} ...`
```

**Review by moreal — APPROVED** [review]
Though it was already merged, I left this review. These comments are **NOT CRITICAL** to merge and use. After merging, please read it once when you're free.

**Pull request merged**